### PR TITLE
fix: Uniformized the code

### LINKED ISSUE #12: [Feature]: Improve the parsing
### Description

Improve the parsing performance and generate the right JSON format :rocket:

### Expected behaviour

You have to:
- Modify the parsing code to uniformize it and parse the Go file only once
- Test your JSON output and think of the format to have a valid OpenAPI specification

Then, you can dig into folder parsing to prepare the next feature :sparkles: 

### Environment

/

### Proposed solution

_No response_

### PR DESCRIPTION
**Pull request by TdeBoynes**
Now it only reads the content 1 time with the new function '"buildHandlersAndEndpoints" in main

Also i left what we had before in coms if there is anything to say about the changes that i made, if it's all good i will delete them

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [x] I have tested this code
- [x] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: src/EndpointParser/parser.go [hunk 1]
```diff
@@ -79,23 +79,23 @@ func parseOneEndpoint(comments []string) EndpointData {
...
+//  */
+// func ParseEndpoint(comments []string) []EndpointData {
+// 	var endpoints []EndpointData
+// 	for index, line := range comments {
+// 		if StartRegexp.MatchString(line) {
+// 			endpoints = append(endpoints, parseOneEndpoint(comments[index+1:]))
+// 		}
+// 	}
+// 	return endpoints
+// }
```
Comment (RezaRahemtola) [inline src/EndpointParser/parser.go]: why was this commented?

File: src/EndpointParser/parser_test.go [hunk 2]
```diff
@@ -1,104 +1,104 @@
...
+// 	if endpoints[2].Headers[0].IsRequired != false {
+// 		t.Errorf("The third endpoint should have header Hello as optional but got %t", endpoints[2].Headers[0].IsRequired)
+// 	}
+// 	if endpoints[2].Headers[1].Key != "no" {
+// 		t.Errorf("The third endpoint should have header no but got %s", endpoints[2].Headers[1].Key)
+// 	}
+// 	if endpoints[2].Headers[1].Description != "test me please" {
+// 		t.Errorf("The third endpoint should have header no with description test me please but got %s", endpoints[2].Headers[1].Description)
+// 	}
+// }
```
Comment (RezaRahemtola) [inline src/EndpointParser/parser_test.go]: same question

File: src/EndpointParser/parser.go (hunk 1 above)
Comment (TdeBoynes) [reply src/EndpointParser/parser.go]: i said why in the description of this PR

File: src/EndpointParser/parser_test.go (hunk 2 above)
Comment (TdeBoynes) [reply src/EndpointParser/parser_test.go]: i said why in the description of this PR

File: src/EndpointParser/parser.go (hunk 1 above)
Comment (RezaRahemtola) [reply src/EndpointParser/parser.go]: Perfect :ok_hand: 

You can fix the tests and deleted them when it's done :+1:

Comment (TdeBoynes) [reply src/EndpointParser/parser.go]: yop my bad on that part

**Pull request merged**